pkg/apiserver/handlers.go

```diff
@@ -45,6 +45,13 @@ var specialVerbs = sets.NewString("proxy", "redirect", "watch")
 // specialVerbsNoSubresources contains root verbs which do not allow subresources
 var specialVerbsNoSubresources = sets.NewString("proxy", "redirect")
 
+// namespaceSubresources contains subresources of namespace
+// this list allows the parser to distinguish between a namespace subresource, and a namespaced resource
+var namespaceSubresources = sets.NewString("status", "finalize")
```

Comment:
In openshift, we split our creation of `map[string]rest.Storage` to be able to keep code like this in sync and ensure that code that needed handle particular resources was able to write a unit test to make sure it doesn't slip.

Can you either do that refactor and test or take an issue to do it?  Otherwise this is going to slip-prone.